processors/minsev/minsev_test.go

```diff
@@ -56,81 +56,68 @@ func (p *processor) ForceFlush(ctx context.Context) error {
 	return p.ReturnErr
 }
 
-func (p *processor) Reset() {
-	p.OnEmitCalls = p.OnEmitCalls[:0]
-	p.EnabledCalls = p.EnabledCalls[:0]
-	p.ShutdownCalls = p.ShutdownCalls[:0]
-	p.ForceFlushCalls = p.ForceFlushCalls[:0]
-}
-
 func TestLogProcessorOnEmit(t *testing.T) {
 	t.Run("Passthrough", func(t *testing.T) {
-		wrapped := &processor{ReturnErr: assert.AnError}
-
-		p := NewLogProcessor(wrapped, api.SeverityTrace1)
-		ctx := context.Background()
-		r := &log.Record{}
 		for _, sev := range severities {
+			wrapped := &processor{ReturnErr: assert.AnError}
+			p := NewLogProcessor(wrapped, api.SeverityTrace1)
+			ctx := context.Background()
+			r := &log.Record{}
+
 			r.SetSeverity(sev)
 			assert.ErrorIs(t, p.OnEmit(ctx, *r), assert.AnError, sev.String())
 
 			if assert.Lenf(t, wrapped.OnEmitCalls, 1, "Record with severity %s not passed-through", sev) {
 				assert.Equal(t, ctx, wrapped.OnEmitCalls[0].Ctx, sev.String())
 				assert.Equal(t, *r, wrapped.OnEmitCalls[0].Record, sev.String())
 			}
-			wrapped.Reset()
 		}
 	})
 
 	t.Run("Dropped", func(t *testing.T) {
-		wrapped := &processor{ReturnErr: assert.AnError}
-
-		p := NewLogProcessor(wrapped, api.SeverityFatal4+1)
-		ctx := context.Background()
-		r := &log.Record{}
 		for _, sev := range severities {
+			wrapped := &processor{ReturnErr: assert.AnError}
+			p := NewLogProcessor(wrapped, api.SeverityFatal4+1)
+			ctx := context.Background()
+			r := &log.Record{}
+
 			r.SetSeverity(sev)
 			assert.NoError(t, p.OnEmit(ctx, *r), assert.AnError, sev.String())
 
-			if !assert.Lenf(t, wrapped.OnEmitCalls, 0, "Record with severity %s passed-through", sev) {
```

Comment:
This if increases the https://en.m.wikipedia.org/wiki/Cyclomatic_complexity

Reply:
I think I gotcha. It seems a bit like the cyclomatic complexity is being over applied here, but it that is truly the desire, there is no need for the `if` statement.

```go
assert.Lenf(t, wrapped.OnEmitCalls, 0, "Record with severity %s passed-through", sev)
wrapped.Reset()
```

That would avoid `if` statement you are concerned about without the added memory inefficiencies. There would still be computation efficiencies as the function call will be a no-op in the existing `true` conditional case.

I'm still not sold that this metric is really serving to better the code with type of change though.